### ISSUE TITLE
new filetype not in the format *.xyz

### ISSUE THREAD

**Issue by JianCheng** [beyondgrep/ack1#198]
Hi,

I am new in ack. 
This is a wonderful tool. Thank you very much.

I normally use cmake, so I would like to define cmake type in .ackrc like this
--type-set=cmake=.cmake,CMakeLists.txt
However, ack automatically adds dot and considers CMakeLists.txt as  .CMakeLists.txt ...
Is it possible to define new filetype for CMakeLists.txt?

The current version of ack only ignores dirs with simple names in --ignore-dir
I really hope ack works like .gitignore.

Thanks

**Comment by petdance** [comment]
You have to do this in your .ackrc

```
--type-set
cmake=.cmake,CMakeLists.txt
```

It's a bug.

We're working on ack 2.0 which will make filetype specifications much easier.

**Comment by JianCheng** [comment]
Thank you for your reply.
I have changed 

```
--type-set=cmake=.cmake,CMakeLists.txt 
```

to

```
--type-set
cmake=.cmake,CMakeLists.txt
```

However when I use `ack --help types`, ack still considers CMakeLists.txt as .CMakeLists.txt

**Comment by petdance** [comment]
I'm sorry, I missed that part.  Yes, you cannot make "CMakeLists.txt" be a --cmake file.  ack 1.x only handles extensions for determining filetypes.

I've just added support for cmake to ack 2.0, though.  The default ackrc now will have these:

```
--type-add=cmake,is,CMakeLists.txt
--type-add=cmake,ext,.cmake
```

**Comment by JianCheng** [comment]
Thanks for your quick reply.
I am glad to see it is supported in ack 2.0.
So for the moment, I can just close this issue.

By the way, I just tried ack 2.0 for a while, and then I was back to 1.96.
It seems that the help file changes a lot. And I tried `ack2 --ackrc=~/.ackrc-2.0 -f` using the new ackrc provided in ack2. 
However --ackrc does not work. 
And when I changed ~/.ackrc as the new one and added `--pager=less -r`, `ack2 -f` showed  all files in a long page, not page by page as usual. 
So I moved back to 1.96.
Anyway, I will keep tracking ack 2.0, and use it once it is released. 
Thank you for providing us such a wonderful tool!!
